### PR TITLE
DEMO-38: Refactor contact

### PR DESCRIPTION
**Pull request by nicolas-bastien**
https://jira.ez.no/browse/DEMO-38

what's in : 
- Centralize sending and in the sender
- Remove email title parameter as this should not be
- Rename thing Message-> Contact in a model folder
- Remove the Form class and call the form type directly in the controller
- add a simple html template for the email
- add form validation

From my point fo view would be great to have a better html template + add a translation story cause even if project is not multilingual it should use translator to handle strings.

### CODE REVIEW

**Comment by nicolas-bastien** [comment]
@bdunogier so I did what we spoke about 20s yesterday.

Now I understand how to handle form action in ez.

**Comment by nicolas-bastien** [comment]
Branche rebased from master

/cc @bdunogier @clash82 @damianz5

File: src/AppBundle/Mail/Sender.php [hunk 1]
```diff
@@ -14,36 +15,47 @@ class Sender
...
      */
-    public function send(Message $message, $title, $senderEmail, $recipientEmail)
+    public function send(Contact $contact)
     {
-        $msg = Swift_Message::newInstance($title, $message->getBody());
-
-        $msg->setFrom($senderEmail)
-            ->setTo($recipientEmail)
-            ->setReplyTo($recipientEmail);
+        $title = 'You have a new message from ' . $contact->getFrom(); //todo handle this with translation : waiting for translation story
```
Comment (nicolas-bastien) [inline src/AppBundle/Mail/Sender.php]: @clash82 I didn't set up translation in this PR cause I want to refactor this too.

I've created a jira for explaining that https://jira.ez.no/browse/DEMO-47.

I think it's better to do it in one PR with all as we already have non translated content in the demo ?

**Comment by nicolas-bastien** [comment]
@clash82 modification done

File: src/AppBundle/Mail/Sender.php [hunk 2]
```diff
@@ -5,45 +5,66 @@
  */
 namespace AppBundle\Mail;
 
-use AppBundle\Entity\Message;
+use AppBundle\Model\Contact;
+use Symfony\Bundle\TwigBundle\TwigEngine as Templating;
 use Swift_Mailer;
 use Swift_Message;
+use Symfony\Component\Translation\TranslatorInterface;
```
Comment (clash82) [inline src/AppBundle/Mail/Sender.php]: we could do small reordering here ;)

File: src/AppBundle/Mail/Sender.php [hunk 3]
```diff
@@ -5,45 +5,66 @@
...
      */
-    public function send(Message $message, $title, $senderEmail, $recipientEmail)
+    public function send(Contact $contact)
     {
-        $msg = Swift_Message::newInstance($title, $message->getBody());
-
-        $msg->setFrom($senderEmail)
-            ->setTo($recipientEmail)
-            ->setReplyTo($recipientEmail);
+        $title = $this->translator->trans('You have a new message from ') . $contact->getFrom();
```
Comment (clash82) [inline src/AppBundle/Mail/Sender.php]: I prefer to use `sprintf` for strings like this but if you want to refactor translations later then it's fine to leave it like that.

`sprintf('You have a new message from %s', $contact->getFrom())`gives more freedom to translators

File: app/Resources/views/full/contact_form.html.twig [hunk 4]
```diff
@@ -39,24 +39,20 @@
...
                                 }
                             }) }}
                         </div>
-                        {{ form_widget(form.submit, {
-                            attr: {
-                                class: 'btn btn-primary btn-lg'
-                            },
-                            label: 'Send'|trans
-                        }) }}
+                        <button type="submit"  class="btn btn-primary btn-lg">{{ 'Send'|trans }}</button>
```
Comment (clash82) [inline app/Resources/views/full/contact_form.html.twig]: CS: there is a small extra space between `"submit"` and `class...` which you can remove ;)

File: src/AppBundle/Mail/Sender.php (hunk 3 above)
Comment (nicolas-bastien) [inline src/AppBundle/Mail/Sender.php]: Your right but in that case it's better to use the translator directly : 

    $title = $this->translator->trans('You have a new message from %from%', ['%from%' => $contact->getFrom()])

Comment (clash82) [inline src/AppBundle/Mail/Sender.php]: Completely agree :)

**Pull request merged**